protocol/bc/byte32.go

```diff
@@ -0,0 +1,89 @@
+package bc
+
+import (
+	"bytes"
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"chain/errors"
+)
+
+type Byte32 [32]byte
```

Comment:
(Clarifying another late-night comment): it's convenient in this PR to put some common code into `byte32`, but the very next PR will reimplement `AssetID` and remove `byte32`.

(Unless you think I should combine the next PR with this one?)